roborock/cloud_api.py

```diff
@@ -78,21 +79,27 @@ def __init__(self, user_data: UserData, device_info: DeviceData) -> None:
         self._encoder: Encoder = create_mqtt_encoder(device_info.device.local_key)
 
     def _mqtt_on_connect(self, *args, **kwargs):
-        _, __, ___, rc, ____ = args
+        rc: ReasonCodes = args[3]
```

Comment:
How about we just make this 

`_mqtt_on_connect(client, userdata: ..., flags: .., reason_code: ..., properties):`